flare_portal/api/views.py

```diff
@@ -47,4 +58,31 @@ def post(self, request: Request, format: str = None) -> Response:
         raise serializers.ValidationError(form.errors)
 
 
+class SubmissionAPIView(APIView):
+    def post(self, request: Request, format: str = None) -> Response:
+        form = ConfigurationForm(request.data)
+
+        if form.is_valid():
+            participant = form.cleaned_data["participant"]
+
+            # Raise error if participant hasn't started
+            if participant.started_at == None:
+                raise serializers.ValidationError(
+                    {"participant": "This participant has not started an experiment."}
+                )
+
+            # Mark participant as finished the experiment
+            if participant.finished_at == None:
+                participant.finished_at = timezone.now()
+                participant.save()
```

Comment:
Suggestion: Business logic should live in `form.save()`.

Reply:
Refactored